### PR TITLE
Updating objects in some of the fluka_funcs functions.

### PR DESCRIPTION
**Pull request by pshriwise**
I took a crack at updating the calls for the PyNE material library in the `fluka_funcs.*` files that were failing to compile, as reported by @ljacobson64  in #712.

The issues with the update from `Tally::particle_name` to `Tally::particle_names` in PyNE is more subtle. What I have in there is likely insufficient. Should those FLUKA routines create multiple tallies if multiple particle names exist? Should that routine fail? Any guidance here is appreciated @bam241 @makeclean @gonuke.

### CODE REVIEW

File: src/fluka/fluka_funcs.cpp
Comment (makeclean) [inline src/fluka/fluka_funcs.cpp]: Originally a pyne::Tally object had a single particle that described it, I take it from the new name that they now have multiple names?

**Review by makeclean — COMMENTED** [review]
Just one query about tally_names?

File: src/fluka/fluka_funcs.cpp
Comment (pshriwise) [reply src/fluka/fluka_funcs.cpp]: That was my understanding too, yeah. Though from these lines in pyne it looks like only single-particle tallies are intended for FLUKA, so maybe what we have here is ok?

https://github.com/pyne/pyne/blob/010c3d3bd2b1530aea4d7e00f2771a129a028673/src/tally.cpp#L600-L603

Hoping for @bam241's thoughts here as it looks like was part of that update.

File: src/fluka/fluka_funcs.cpp
Comment (bam241) [inline src/fluka/fluka_funcs.cpp]: why not pass the material library directly as you can use it as a std::map

File: src/fluka/fluka_funcs.cpp
Comment (bam241) [inline src/fluka/fluka_funcs.cpp]: if you keep the pyne::MaterialLibrary you could use get_material method and or the [] operator

File: src/fluka/fluka_funcs.cpp
Comment (bam241) [inline src/fluka/fluka_funcs.cpp]: I am not 100% on what is done there.
but with the Pyne::MaterialLibrary you can use `get_nuclist()` to get a set of the nuclides (if required...)

File: src/fluka/fluka_funcs.cpp
Comment (pshriwise) [reply src/fluka/fluka_funcs.cpp]: Are you saying we should leave the function arguments as they were and pass this the `mat_lib` variable?

Comment (bam241) [reply src/fluka/fluka_funcs.cpp]: I think we should pass directly the material_library object
```suggestion
    fludag_all_materials(material_str, workflow_data.material_library);
```

Comment (pshriwise) [reply src/fluka/fluka_funcs.cpp]: Ah, nevermind. I see what you're saying. Changes pending...

File: src/fluka/fluka_funcs.cpp
Comment (makeclean) [reply src/fluka/fluka_funcs.cpp]: well thats all that does, but that functionality from materialibrary must be new?

Comment (pshriwise) [reply src/fluka/fluka_funcs.cpp]: Yeah I'm honestly not sure about what's being done here either. In terms of the objects, it's iterating over the material objects and adding them. `nuc` is an odd variable name though. You ok with a change to `it` as well here?

Comment (makeclean) [reply src/fluka/fluka_funcs.cpp]: what it does is use pyne::material addition to build a material object that has every nuclide present in the entire problem.

Comment (makeclean) [reply src/fluka/fluka_funcs.cpp]: that 'special' unique material then does collapse_elements() to get a material that contains the special set of nuclides that FLUKA is aware of

Comment (makeclean) [reply src/fluka/fluka_funcs.cpp]: but that material object is never written

Comment (makeclean) [reply src/fluka/fluka_funcs.cpp]: nuc should be material or mat or something like that

**Review by bam241 — APPROVED** [review]
Thx @pshriwise for addressing those.
LGTM!

File: src/fluka/fluka_funcs.cpp
Comment (bam241) [reply src/fluka/fluka_funcs.cpp]: > makeclean 
> well thats all that does, but that functionality from materialibrary must be new?

Yes it has been added when we created the C++ API

Comment (pshriwise) [reply src/fluka/fluka_funcs.cpp]: Done!

**Comment by pshriwise** [comment]
This looking ok now @bam241 @makeclean?

**Comment by bam241** [comment]
Fine by me!
Waiting for @makeclean approval to merge !

**Comment by bam241** [comment]
LGTM!
In order to keep this release active, I'll be merging!
If additional comments rise, they will be addressed in an other PR :)

**Comment by bam241** [comment]
thx @pshriwise for addressing those issues.

**Pull request merged**
